packages/js/product-editor/src/components/number-control/number-control.tsx

```diff
@@ -47,164 +50,175 @@ export type NumberProps = {
 const MEDIUM_DELAY = 500;
 const SHORT_DELAY = 100;
 
-export const NumberControl: React.FC< NumberProps > = ( {
-	value,
-	onChange,
-	label,
-	suffix,
-	help,
-	error,
-	onBlur,
-	onFocus,
-	required,
-	tooltip,
-	placeholder,
-	disabled,
-	step = 1,
-	min = -Infinity,
-	max = Infinity,
-}: NumberProps ) => {
-	const id = useInstanceId( BaseControl, 'product_number_field' ) as string;
-	const [ isFocused, setIsFocused ] = useState( false );
-	const unfocusIfOutside = ( event: React.FocusEvent ) => {
-		if (
-			! document
-				.getElementById( id )
-				?.parentElement?.contains( event.relatedTarget )
-		) {
-			setIsFocused( false );
-			onBlur?.();
-		}
-	};
-	const inputProps = useNumberInputProps( {
-		value: value || '',
-		onChange,
-		onFocus: () => {
-			setIsFocused( true );
-			onFocus?.();
-		},
-		min,
-		max,
-	} );
+export const NumberControl: React.FC< NumberProps > = forwardRef(
+	(
+		{
+			identifier,
+			value,
+			onChange,
+			label,
+			suffix,
+			help,
+			error,
+			onBlur,
+			onFocus,
+			required,
+			tooltip,
+			placeholder,
+			disabled,
+			step = 1,
+			min = -Infinity,
+			max = Infinity,
+		}: NumberProps,
+		ref: ForwardedRef< HTMLInputElement >
+	) => {
+		const instanceId = useInstanceId(
+			BaseControl,
+			'product_number_field'
+		) as string;
+		const id = identifier ?? instanceId;
+		const [ isFocused, setIsFocused ] = useState( false );
+		const unfocusIfOutside = ( event: React.FocusEvent ) => {
+			if (
+				! document
+					.getElementById( id )
+					?.parentElement?.contains( event.relatedTarget )
+			) {
+				setIsFocused( false );
+				onBlur?.();
+			}
+		};
+		const inputProps = useNumberInputProps( {
+			value: value || '',
+			onChange,
+			onFocus: () => {
```

Comment:
That line was affected by the linter, because I added a function `forwardRef` [on top](https://github.com/woocommerce/woocommerce/pull/49357/files#diff-36fe6f86d720a06f7d285658f2405bb26704e6f3cc81b621e874ca950f5cdc68R53). I like the idea of getting ride of not necessary arrow functions but I would tackle those changes in a follow-up PR.

Reply:
Done in the commit [f9f38dc](https://github.com/woocommerce/woocommerce/pull/49357/commits/f9f38dcf79b05c57e106924f07a3459e984b9e12)